src/api/adamik/history.ts

```diff
@@ -7,34 +7,30 @@ interface AccountHistoryResponse {
   chainId: string;
   accountId: string;
   transactions: FinalizedTransaction[];
+  pagination?: {
+    nextPage: string | null;
+  };
 }
 
 export const getAccountHistory = async (
```

Comment:
This function correctly sets `nextPage` if provided, however it doesn't read it from the response and does not loop while there is a `nextPage` token.

You just need the same as for validators: the `do {} while` loop to iterate as long as a `nextPage` is provided.

Reply:
It's kind of on purpose... 

1. I only fetch the first page
2. fetch the other page only when the user click on Load More

This avoids unnecessary calls.